### PR TITLE
Load textures

### PR DESCRIPTION
**Pull request by Giggzzz**
# Load textures

* Load all the textures at init()
* Fallback to defaults values if a path is invalid or undefined
* Free all images and paths on close()
* Create images for floor and ceiling based on a color
* Convert RGB string (255,255,255) to an int representing the color

### CODE REVIEW

File: assets/textures/default_ceil.xpm [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/* XPM */
+static char * wall_xpm[] = {
```
Comment (t-h2o) [inline assets/textures/default_ceil.xpm:2]: This is the same name of the file
```suggestion
static char * default_ceil[] = {
```

File: headers/cub3d.h [hunk 2]
```diff
@@ -74,13 +83,31 @@ enum e_mm {
...
 	char	*south;
+	void	*img_south;
 	char	*east;
+	void	*img_east;
 	char	*west;
+	void	*img_west;
 	char	*floor;
+	void	*img_floor;
 	char	*ceil;
+	void	*img_ceil;
```
Comment (t-h2o) [inline headers/cub3d.h]: Do we do a better structure ?

File: srcs/close.c [hunk 3]
```diff
@@ -6,32 +6,63 @@
...
+	if (info->texture.img_south)
+		mlx_destroy_image(info->mlx[INIT], info->texture.img_south);
+	if (info->texture.img_east)
+		mlx_destroy_image(info->mlx[INIT], info->texture.img_east);
+	if (info->texture.img_west)
+		mlx_destroy_image(info->mlx[INIT], info->texture.img_west);
+	if (info->texture.img_floor)
+		mlx_destroy_image(info->mlx[INIT], info->texture.img_floor);
+	if (info->texture.img_ceil)
+		mlx_destroy_image(info->mlx[INIT], info->texture.img_ceil);
```
Comment (t-h2o) [inline srcs/close.c]: Do we create a file `utils_mlx.c` ?
with function like:
```c
my_destroy_image(void *init, void *img)
{
    if (img)
        mlx_destroy_image(init, img);
}

File: srcs/close.c [hunk 4]
```diff
@@ -6,32 +6,63 @@
...
+	if (info->texture.south)
+		free(info->texture.south);
+	if (info->texture.east)
+		free(info->texture.east);
+	if (info->texture.west)
+		free(info->texture.west);
+	if (info->texture.floor)
+		free(info->texture.floor);
+	if (info->texture.ceil)
+		free(info->texture.ceil);
```
Comment (t-h2o) [inline srcs/close.c]: It's useless because we can do `free(NULL)`, free do not report any error

File: srcs/main.c [hunk 5]
```diff
@@ -20,13 +20,10 @@ int	main(int ac, char **av)
 	ft_putstr("Cub3D\n");
 
 	if (ac != 2)
-		exit_error("Invalid argument");
+		exit(error_msg("Invalid argument"));
```
Comment (t-h2o) [inline srcs/main.c:23]: Nice !

**Review by t-h2o — COMMENTED** [review]
My first review

File: assets/textures/default_ceil.xpm (hunk 1 above)
Comment (Giggzzz) [reply assets/textures/default_ceil.xpm:2]: These textures are just temporary and will never be displayed. We will replace them of course

File: headers/cub3d.h (hunk 2 above)
Comment (Giggzzz) [reply headers/cub3d.h]: Yes we do !

File: srcs/close.c (hunk 3 above)
Comment (Giggzzz) [reply srcs/close.c]: Good idea! Let's do it

File: srcs/close.c (hunk 4 above)
Comment (Giggzzz) [reply srcs/close.c]: Good observation. I will fix it

File: headers/cub3d.h (hunk 2 above)
Comment (Giggzzz) [reply headers/cub3d.h]: done : 00868b206bc446c663fd618ddf212e01e5071c5a

File: srcs/close.c (hunk 3 above)
Comment (Giggzzz) [reply srcs/close.c]: done: 8c121ed0a3121ccb0c18d25d4b5ec7b24d1265b6

File: srcs/close.c (hunk 4 above)
Comment (Giggzzz) [reply srcs/close.c]: done : 8c121ed0a3121ccb0c18d25d4b5ec7b24d1265b6

**Review by t-h2o — APPROVED** [review]
Let's go

**Pull request merged**